gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -924,6 +926,227 @@ public void deleteObjects(List<StorageResourceId> fullObjectNames) throws IOExce
     }
   }
 
+  /** See {@link GoogleCloudStorage#deleteFolders(List)} for details about expected behavior. */
+  @Override
+  public void deleteFolders(List<FolderInfo> folders) throws IOException {
+
+    if (folders.isEmpty()) {
+      return;
+    }
+
+    // Gather exceptions to wrap in a composite exception at the end.
+    final KeySetView<IOException, Boolean> innerExceptions = ConcurrentHashMap.newKeySet();
+    String traceContext = String.format("batchFolderDelete(size=%s)", folders.size());
+    try (ITraceOperation to = TraceOperation.addToExistingTrace(traceContext)) {
+
+      // threads for parallel delete of folder resources
+      BatchHelper batchHelper =
+          batchFactory.newBatchHelper(
+              httpRequestInitializer,
+              storage,
+              storageOptions.getMaxRequestsPerBatch(),
+              folders.size(),
+              storageOptions.getBatchThreads(),
+              "batchFolderDelete");
+
+      // Map to store number of children for each parent object
+      HashMap<String, Long> occurenceOfFolder = new HashMap<>();
+      for (FolderInfo currentFolder : folders) {
+        // inserting each folder
+        if (!occurenceOfFolder.containsKey(currentFolder)) {
+          occurenceOfFolder.put(currentFolder.getFolderName(), 0L);
+        }
+
+        String parentFolder = currentFolder.getParentFolderName();
+
+        if (parentFolder != "" && parentFolder != null) {
+          occurenceOfFolder.put(
+              parentFolder,
+              (occurenceOfFolder.containsKey(parentFolder)
+                      ? occurenceOfFolder.get(parentFolder)
+                      : 0)
+                  + 1);
+        }
+      }
+
+      BlockingQueue<FolderInfo> folderDeleteBlockingQueue = new LinkedBlockingQueue<>();
+      for (FolderInfo currentFolder : folders) {
+        if (occurenceOfFolder.get(currentFolder.getFolderName()) == 0L) {
+          addFolderResourceInBlockingQueue(
+              currentFolder, folderDeleteBlockingQueue, innerExceptions);
+        }
+      }
+
+      while (!occurenceOfFolder.isEmpty() && innerExceptions.isEmpty()) {
+        while (!folderDeleteBlockingQueue.isEmpty()) {
+          FolderInfo folderToDelete = null;
+          try {
+            folderToDelete = folderDeleteBlockingQueue.poll(1, TimeUnit.MINUTES);
```

Comment:
I do not think this is the right thing to do. This will block for 1 minute. Why not loop for folder.size()?

Reply:
I updated it by removing the buffer of 1 minute. I accidently forgot to remove it after adding the top while loop which is checking for exceptions and map